src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -0,0 +1,124 @@
+/*
+  Copyright (c) 2021, RTE (http://www.rte-france.com)
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.repositories;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.powsybl.commons.PowsyblException;
+import org.gridsuite.modification.server.ModificationType;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.ElementaryModificationInfos;
+import org.gridsuite.modification.server.dto.ModificationInfos;
+import org.gridsuite.modification.server.entities.*;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.MODIFICATION_GROUP_NOT_FOUND;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ */
+@Repository
+public class NetworkModificationRepository {
+    private ModificationGroupRepository modificationGroupRepository;
+
+    private ModificationRepository modificationRepository;
+
+    public NetworkModificationRepository(ModificationGroupRepository modificationGroupRepository, ModificationRepository modificationRepository) {
+        this.modificationGroupRepository = modificationGroupRepository;
+        this.modificationRepository = modificationRepository;
+    }
+
+    @Transactional // To have the 2 delete in the same transaction (atomic)
+    public void deleteAll() {
+        modificationRepository.deleteAll();
+        modificationGroupRepository.deleteAll();
+    }
+
+    @Transactional // To have the 2 create in the same transaction (atomic)
```

Comment:
don't forget that transactions also affect selects !

Reply:
How ?

Reply:
The semantics are defined by isolation levels, but for example for REPEATABLE_READS if you wrap your function in a transaction you will get the value at the begining of the function, whereas if you don't wrap in a transaction you will get the value "at the moment of the select". Many more complex scenarios are possible..
I was just saying that because the comment says "2 inserts", but there is also 1 select in this function

Reply:
Yes but is not this what we want to avoid the group being deleted by another transaction